iceoryx2-pal/posix/src/qnx/types.rs

```diff
@@ -91,11 +90,13 @@ pub type pthread_rwlockattr_t = sync_attr_t;
 pub type pthread_rwlock_t = crate::internal::pthread_rwlock_t;
 impl MemZeroedStruct for pthread_rwlock_t {}
 
-pub type pthread_mutex_t = sync_t;
+pub type pthread_mutex_t = crate::internal::pthread_mutex_t;
+impl MemZeroedStruct for pthread_mutex_t {}
 
 pub type pthread_mutexattr_t = sync_attr_t;
 
-pub type sem_t = sync_t;
+pub type sem_t = crate::internal::sem_t;
```

Comment:
You will probably need to do something like this for these changes:

```suggestion
#[cfg(target_env = "nto71")]
pub type sem_t = sync_t;
#[cfg(target_env = "nto80")]
pub type sem_t = crate::internal::sem_t;
```